src/range_proof.rs

```diff
@@ -0,0 +1,558 @@
+}
+
+impl RangeProof {
+    /// The maximum bit length that proofs can be generated for
+    pub const MAX_BIT_LENGTH: usize = 64;
+
+    fn get_a(&self) -> Result<RistrettoPoint, ProofError> {
+        self.a.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'a' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_a1(&self) -> Result<RistrettoPoint, ProofError> {
+        self.a1.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'a1' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_b(&self) -> Result<RistrettoPoint, ProofError> {
+        self.b.decompress().ok_or_else(|| {
+            ProofError::InternalDataInconsistent("Member 'b' was not the canonical encoding of a point".to_string())
+        })
+    }
+
+    fn get_r1(&self) -> Scalar {
+        self.r1
+    }
+
+    fn get_s1(&self) -> Scalar {
+        self.s1
+    }
+
+    fn get_d1(&self) -> Scalar {
+        self.d1
+    }
+
+    fn get_li(&self) -> Result<Vec<RistrettoPoint>, ProofError> {
+        if !self.li.is_empty() {
+            let mut li = Vec::with_capacity(self.li.len());
+            for item in self.li.clone() {
+                li.push(item.decompress().ok_or_else(|| {
+                    ProofError::InternalDataInconsistent(
+                        "An item in member 'L' was not the canonical encoding of a point".to_string(),
+                    )
+                })?)
+            }
+            Ok(li)
+        } else {
+            Err(ProofError::InternalDataInconsistent(
+                "Vector 'L' not assigned yet".to_string(),
+            ))
+        }
+    }
+
+    fn get_ri(&self) -> Result<Vec<RistrettoPoint>, ProofError> {
+        if !self.li.is_empty() {
+            let mut ri = Vec::with_capacity(self.ri.len());
+            for item in self.ri.clone() {
+                ri.push(item.decompress().ok_or_else(|| {
+                    ProofError::InternalDataInconsistent(
+                        "An item in member 'R' was not the canonical encoding of a point".to_string(),
+                    )
+                })?)
+            }
+            Ok(ri)
+        } else {
+            Err(ProofError::InternalDataInconsistent(
+                "Vector 'R' not assigned yet".to_string(),
+            ))
+        }
+    }
+
+    /// Create a single or aggregated range proof for a single party that knows all the secrets
+    pub fn prove(
+        transcript: &mut Transcript,
+        statement: &RangeStatement,
+        witness: RangeWitness,
+    ) -> Result<RangeProof, ProofError> {
+        let batch_size = statement.commitments.len();
+        if witness.openings.len() != batch_size {
+            return Err(ProofError::InvalidLength(
+                "Invalid range statement - commitments and openings do not match!".to_string(),
+            ));
+        }
+        for j in 0..batch_size {
```

Comment:
This step seems superfluous. The caller either has valid data or can do this check themselves.

If it must be done, then a single `multiscalar_mult` would be better